### ISSUE TITLE
Changing tags of azurerm_search_service forces new resource

### ISSUE THREAD

**Issue by maxbog** [terraform-providers/terraform-provider-azurerm#3641]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.12.1
+ provider.azurerm v1.30.1

### Affected Resource(s)

* `azurerm_search_service`

### Terraform Configuration Files

```hcl
resource "azurerm_search_service" "common" {
  location = var.location
  resource_group_name = azurerm_resource_group.services.name

  name = "example-search"
  sku = "standard"

  replica_count = 1
  partition_count = 1

  tags = {
    tag1 = "value1"
  }
}
```

changed to: 

```hcl
resource "azurerm_search_service" "common" {
  location = var.location
  resource_group_name = azurerm_resource_group.services.name

  name = "example-search"
  sku = "standard"

  replica_count = 1
  partition_count = 1

  tags = {
    tag1 = "value2"
  }
}
```

### Expected Behavior

`tag1` value should have changed without affecting the resource

### Actual Behavior

terraform wants to recreate the search service to change the tag

### Steps to Reproduce

1. Change a value of any `azurerm_search_service` tag
2. `terraform apply`

### References

* https://github.com/terraform-providers/terraform-provider-azurerm/blob/41f071c23b449df7f4dfe9aa57ac81c3386e9e04/azurerm/resource_arm_search_service.go#L73

Why does the referenced line have `tagsForceNewSchema` applied instead of `tagsSchema`?

**Comment by maxbog** [comment]
Ok, I tried to fix it by changing the schema, but unfortunately, when I try to update a tag value, I get an error:
```
azurerm_search_service.test: Modifying... [id=/subscriptions/***/resourceGroups/acctest-search/providers/Microsoft.Search/searchServices/ttestsearch]

Error: doesn't support update

  on test.tf line 10, in resource "azurerm_search_service" "test":
  10: resource "azurerm_search_service" "test" {
```

However, I can easily update the resource using the REST API. Maybe this is some issue with the Azure SDK or something else in the provider code that could cause that error?

**Comment by maxbog** [comment]
I created a PR with the change, which unfortunately does not pass acceptance tests :(. Any help is very welcome.

**Comment by tombuildsstuff** [comment]
Fixed via #3660 - thanks @maxbog :)
